taf/developer_tool.py

```diff
@@ -770,6 +770,25 @@ def export_yk_certificate(certs_dir, key):
         f.write(yk.export_piv_x509())
 
 
+def export_targets_history(repo_path, commit=None, output=None, target_repos=None):
+    auth_repo = AuthenticationRepo(repo_path)
+    commits = auth_repo.all_commits_since_commit(commit, branch="master")
+    if not len(target_repos):
+        target_repos = None
+    commits_on_branches = auth_repo.sorted_commits_and_branches_per_repositories(
+        commits, target_repos
+    )
+    commits_json = json.dumps(commits_on_branches, indent=4)
+    if output is not None and not output.endswith(".json"):
+        output += ".json"
+    if output is not None:
+        output = Path(output).resolve()
```

Comment:
You can move `if not output.endswith(".json"):` to line 785, or check it with `output.suffix != '.json'` at line 786.